x-pack/plugin/sql/jdbc/src/main/java/org/elasticsearch/xpack/sql/jdbc/Debug.java

```diff
@@ -113,17 +113,16 @@ private static DebugLog logger(JdbcConfiguration info, PrintWriter managedPrinte
             synchronized (Debug.class) {
                 log = OUTPUT_MANAGED.get(managedPrinter);
                 if (log == null) {
-                    log = new DebugLog(managedPrinter);
+                    log = new DebugLog(managedPrinter, info.debugBuffered());
                     OUTPUT_MANAGED.put(managedPrinter, log);
                 }
-                return log;
```

Comment:
The method changes make it more complicated and hard to follow - there's this `log ==null` being repeated which begs the question, what happens it if it initialized?
I think the method is good as it is - it creates a logger or returns one that already exists. Either wrap this method in another one that calls `logSystemInfo` (rename the existing method into something like createLogger and then logger calls `createLogger().logSystemInfo` or wrap that at the consumer site - inside `proxy`.

However currently the systemInfo is being called per each `Connection` which I think is excessive - we want the system information to be once per log, at the beginning.
In which case, the system info should called when a new log is created, essentially after each `new DebugLog`.
To keep things incapsulated, instead of calling the constructor, one could call a method `createLog` which internally calls `new DebugLog` and right after calls logSystemInfo.
This way any other initialization that would need to occur, would happen in that method regardless of the actual method taking place and only _once_ per logger.
It might make sense to call `flush` after logging the system info.